### DISCUSSION TITLE
Is ipxe  http download working on a vmware guest in bios boot mode and a vmxnet3 card

### DISCUSSION THREAD

**Discussion by ptempier** [ipxe/ipxe#1287] [Q&A]
Hello

If someone could please test this specific combination and report, it takes 2 minutes if your ipxe setup is working and you have vmware.

I am trying to diagnose this since 7-10 days. 
I thought it was an issue with the firewalls or ipxe tcp stack.
But i was always doing my tests on the same vm.

The problem went away with physical computers or by switching to an e1000 card or uefi boot. 
(btw, uefi boot has documented issue that pxe boot works only if you dont modify the network cards after having created the guest).

Thats's on esxi 7.0u2 if that makes any différence.
Quite not sure if the issue is from vmware vmxnet3 card or ipxe driver for that card.

To trigger the issue the most simple is to go the ipxe shell , type :
```
ctrl-b during boot
dhcp
initrd http:://whatever && imgfree
```
.. wait for download to finish , arrow up , enter, rince and repeat 20 times.
At one point, the download will get stuck at some random percentage.

As i understand its a very old issue, i've found bug report dating from 2014-2016.
Unless it s mixed with other bugs with the same symptôme.
Redhat published a fix for one of their product, which was to timeout the download in the ipxe script then try it again. ...

Still need mroe tests to confirm, but when booting pxelinux cd from ipxe http san,
the boot also hang at random time.
i think the image is fully loaded, like for linux it downlaod the initrd.
The initrd is about 120M, so it won t always hang, but the windows pe image is 1G so 99% chances it ll hang.
The fix is the same, switching the network card from vmxnet3 to e1000 solves the issue.

**Comment by stappersg** [comment]
> please test this specific combination

:smiley:  please be more specific, like
* stating the size of the `initrd` file
* telling how many hops the HTTP server is away
* sharing information about the HTTP server


> and report

That _might_ bring a "works for me" or a "Yeah, one in twenty of the attemps fails", none of such reports brings the iPXE project further.


Acknowledge on the _"I am trying to diagnose this since 7-10 days"_, I hope original poster gets beyond the _"HTTP download stalls occasionally"_.
